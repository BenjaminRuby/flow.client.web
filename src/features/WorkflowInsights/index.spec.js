import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import { WorkflowInsights } from "./index";

jest.mock("Components/NavigateBack", () => "NavigateBack");
jest.mock("./CustomAreaChart", () => "CustomAreaChart");
jest.mock("./CustomScatterChart", () => "CustomScatterChart");
jest.mock("./CustomPieChart", () => "CustomPieChart");
jest.mock("./WidgetCard", () => "WidgetCard");
jest.mock("Components/SearchFilterBar", () => "SearchFilterBar");
jest.mock("Components/SimpleSelectFilter", () => "SimpleSelectFilter");

const location = {};
const insightsActions = {
  fetch: () => new Promise(resolve => resolve({ test: "test" }))
};
const teamsActions = {
  fetch: () => new Promise(() => {})
};
const teams = {
  isFetching: false,
  status: "success",
  error: "",
  data: []
};
const insights = {
  isFetching: false,
  status: "success",
  error: "",
  data: {
    totalActivitiesExecuted: 107,
    medianExecutionTime: 24435,
    executions: [
      {
        creationDate: 1542292277199,
        duration: 20825,
        id: "5bed8335b336a5000103aa2b",
        initiatedByUserId: "5bbcdd38ee763e00011f5b72",
        initiatedByUserName: "Adrienne Hudson",
        status: "completed",
        workflowId: "5bed82f6b336a5000103aa1d",
        workflowRevisionid: "5bed82f7b336a5000103aa20",
        description: "test delete",
        icon: "flow",
        shortDescription: "test",
        workflowName: "Adrienne3",
        teamName: "CAI Offering Team"
      }
    ]
  }
};

describe("WorkflowInsights --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowInsights", () => {
    const renderedValue = renderer
      .create(
        <MemoryRouter>
          <WorkflowInsights
            insights={insights}
            teams={teams}
            insightsActions={insightsActions}
            teamsActions={teamsActions}
            location={location}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("WorkflowInsights --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <WorkflowInsights
          insights={insights}
          teams={teams}
          insightsActions={insightsActions}
          teamsActions={teamsActions}
          location={location}
        />
      </MemoryRouter>
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
