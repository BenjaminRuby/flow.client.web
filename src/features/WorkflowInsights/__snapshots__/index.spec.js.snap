// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowInsights --- Snapshot Capturing Snapshot of WorkflowInsights 1`] = `
<div
  className="c-workflow-insights"
>
  <nav
    className="s-workflow-insights-navigation"
  >
    <NavigateBack
      text="Back to Workflows"
      to="/workflows"
    />
  </nav>
  <div
    className="c-workflow-insights-header"
  >
    <div
      aria-label="Choose an item"
      className="bx--dropdown bx--list-box"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="listbox"
      tabIndex={null}
    >
      <div
        aria-expanded={false}
        aria-haspopup={true}
        aria-label="open menu"
        className="bx--list-box__field"
        data-toggle={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
        type="button"
      >
        <span
          className="bx--list-box__label"
          htmlFor="downshift-0-input"
        >
          All Teams
        </span>
        <div
          className="bx--list-box__menu-icon"
        >
          <svg
            alt="Open menu"
            aria-label="Open menu"
            fillRule="evenodd"
            focusable="false"
            height="5"
            role="img"
            viewBox="0 0 10 5"
            width="10"
          >
            <title>
              Open menu
            </title>
            <path
              d="M0 0l5 4.998L10 0z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      aria-label="Choose an item"
      className="bx--dropdown bx--list-box"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="listbox"
      tabIndex={null}
    >
      <div
        aria-expanded={false}
        aria-haspopup={true}
        aria-label="open menu"
        className="bx--list-box__field"
        data-toggle={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
        type="button"
      >
        <span
          className="bx--list-box__label"
          htmlFor="downshift-1-input"
        >
          All Workflows
        </span>
        <div
          className="bx--list-box__menu-icon"
        >
          <svg
            alt="Open menu"
            aria-label="Open menu"
            fillRule="evenodd"
            focusable="false"
            height="5"
            role="img"
            viewBox="0 0 10 5"
            width="10"
          >
            <title>
              Open menu
            </title>
            <path
              d="M0 0l5 4.998L10 0z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      aria-label="Choose an item"
      className="bx--dropdown bx--list-box"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="listbox"
      tabIndex={null}
    >
      <div
        aria-expanded={false}
        aria-haspopup={true}
        aria-label="open menu"
        className="bx--list-box__field"
        data-toggle={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
        type="button"
      >
        <span
          className="bx--list-box__label"
          htmlFor="downshift-2-input"
        >
          3 months
        </span>
        <div
          className="bx--list-box__menu-icon"
        >
          <svg
            alt="Open menu"
            aria-label="Open menu"
            fillRule="evenodd"
            focusable="false"
            height="5"
            role="img"
            viewBox="0 0 10 5"
            width="10"
          >
            <title>
              Open menu
            </title>
            <path
              d="M0 0l5 4.998L10 0z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c-workflow-insights-stats-widgets"
  >
    <WidgetCard
      title="Total Executed"
      type="stat"
    >
      <p
        className="b-workflow-insights__stats-label --no-data"
      >
        No Data
      </p>
    </WidgetCard>
    <WidgetCard
      title="Median Duration"
      type="stat"
    >
      <p
        className="b-workflow-insights__stats-label --no-data"
      >
        No Data
      </p>
    </WidgetCard>
    <WidgetCard
      title="Success Rate"
      type="stat"
    >
      <p
        className="b-workflow-insights__stats-label --no-data"
      >
        No Data
      </p>
    </WidgetCard>
  </div>
  <div
    className="c-workflow-insights-graphs-widgets"
  >
    <WidgetCard
      title="Executions"
      type="graph"
    >
      <p
        className="b-workflow-insights__graphs-label --no-data"
      >
        No Data
      </p>
    </WidgetCard>
    <WidgetCard
      title="Execution Time"
      type="graph"
    >
      <p
        className="b-workflow-insights__graphs-label --no-data"
      >
        No Data
      </p>
    </WidgetCard>
  </div>
</div>
`;
